### ISSUE TITLE
Can left/right arrow keys not skip four spaces when inside pipe tables?

### ISSUE THREAD

**Issue by bcmadsen** [RickStrahl/MarkdownMonster#728]
In the edit window, if you type a long string of spaces, and then if you use the left and right arrows to move around on that line, the arrow keys are designed to skip four spaces at a time, so groups of four spaces are treated as pseudo-tab columns.

I'm not actually complaining about that, except when I'm inside a pipe table. At work I keep track of hours worked on each project with a table that starts out looking like this:

```
| Project                      | Mon | Tue | Wed | Thu | Fri |
|:-----------------------------|:---:|:---:|:---:|:---:|:---:|
|                              |     |     |     |     |     |
|                              |     |     |     |     |     |
|                              |     |     |     |     |     |
```

(In case you're wondering, no, this isn't mandated by the company, nor is it a report I submit to anybody. This is just my daily work journal that I keep as a personal record of what I've worked on.)

When I'm entering hours worked into this table, I generally like to leave the pipes where they are so that they stay lined up like this, and so I set the cursor to overwrite rather than insert and I use arrows to place numbers of hours worked precisely into the right locations.

And when I'm doing this, it becomes somewhat disorienting to have the arrow keys jump more than a single space at a time. It's not what I expect when I'm trying to fine-tune specific values in specific rows and columns.

Do I have any way to turn off that four-column skip behavior altogether? I don't actually want to disable it, most of the time. Most of the time I like it. But if I can toggle that behavior globally, maybe that's a (drastic?) workaround. Ideally though, what do you think about the idea of just not skipping four spaces at a time whenever my cursor is in between the borders of a pipe table? That would give me exactly what I really want, to be honest.

I totally admit that this is a nitpick and a minor complaint, but I often wish I had some control over this, at least in this particular situation.

**Comment by RickStrahl** [comment]
Hmmm... yes I see what you mean. This appears to be a core issue with ACE Editor as you can see if you paste your table into the ACE Editor Kitchen Sink sample:

https://ace.c9.io/build/kitchen-sink.html

It has the exact same behavior. 

Could you raise the issue on that repo? MM uses ACE Editor as the core editor and while there are some minor enhancements on top of ACE that we add, it's relatively minor and we depend on those ACE features.  Note if you do this would pertain specifically to the **Markdown Syntax** which is built into the editor.

**Comment by RickStrahl** [comment]
Re-Posted  on the ACE Editor issue tracker

https://github.com/ajaxorg/ace/issues/4419

**Comment by RickStrahl** [comment]
Heard back from @nightwing on the ACE repo and for now I have a workaround that disables softtab skipping, which is what is apparently causing this issue. This changes behaviors in the editor where arrow left/right skips to the next tab.

I'm adding this workaround which disables automatic soft tab expansion - ie. it jumps to the next tab stop with arrow keys if characters between the tab stop and cursor are all spaces. Looks like VS Code, VS, Rider, WebStorm etc. all do that as the default behavior so this is actually more appropriate anyway.

Fixed in c459ac57
